lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -257,7 +257,7 @@ instance Arbitrary GenTxHistory where
         -- We discard pending transaction from any 'GenTxHistory since,
         -- inserting a pending transaction actually has an effect on the
         -- checkpoint's pending transactions of the same wallet.
-        filter (not . isPending . snd) <$> scale (`mod` 100) arbitrary
+        filter (not . isPending . snd) <$> scale (min 25) arbitrary
```

Comment:
Just curious about the choice to use `min` here, rather than `mod`.

If the generator frequently generates value lengths that are greater than 25 (according to some distribution), wouldn't this mean that we end up choosing 25 for much of the time?

Reply:
Important to realise: The size parameter is not the length of the generated list - it's the upper bound on list length.

QuickCheck gradually increases the size parameter as it runs tests.

The `scale` function adjusts the size according to the given (non-)linear function.
Using `mod` caps the size parameter at 100, but it doesn't make sense to me.
Better to clamp the size with the `min` function.

There is also the `resize` function which fixes a constant size. But I would like the size to grow - up to a point.